### PR TITLE
Add circuits table

### PR DESCRIPTION
**Pull request by eloaverona**
This adds the circuits table to the main page of the circuit sapling

#### To test
1. Run:
```
docker-compose -f docker/docker-compose up --build
```
2. To see the alpha node UI go to: `localhost:3030`

3. Click on the `circuits` option in the left side menu, it will forward you to  `localhost:3030/circuits`: 
Screen Shot 2020-04-20 at 8 55 25 AM

4. Click the table columns and check that they sort the row as expected. Below is the example of sorting the table by comments in ascendant order 
Screen Shot 2020-04-20 at 8 56 22 AM

5. Check on the `status` column header and check that you can filter the circuits by status. Below is the result of filtering by "Action required"
Screen Shot 2020-04-20 at 8 58 00 AM
Screen Shot 2020-04-20 at 8 58 20 AM

6.  Check that filtering by search term and status show the circuits in intersection of the two filters. Below is the result of filtering by status "action required" and the search term "gameroom":
Screen Shot 2020-04-20 at 9 01 32 AM

7. Perform any other tests you can think of, let me know of any bugs or edge cases!

### CODE REVIEW

**Review by vaporos — DISMISSED** [review]
Tested.

**Review by agunde406 — DISMISSED** [review]
Tested locally

File: app/src/theme/colors.scss [hunk 1]
```diff
@@ -30,6 +30,7 @@ $color-secondary-dark: darken($color-secondary, 15%);
   --color-secondary-dark: #{$color-secondary-dark};
   --color-dark-grey: #333333;
   --color-grey: #555555;
+  --color-light-grey: #DDDDDD;
```
Comment (dplumb94) [inline app/src/theme/colors.scss:33]: Can you contribute this back to the design system in CanopyJS? I don't think you need to overwrite some of these values if you are importing that css.

File: saplings/circuits/src/components/circuitsTable/CircuitsTable.scss [hunk 2]
```diff
@@ -21,5 +21,37 @@
...
+    th, td {
+      border-bottom: 1px solid var(--color-light-grey);
+      padding: 2rem 2rem 2rem 0;
+      max-width: 13rem;
+    }
+
+    .table-header th {
+      color: var(--color-grey);
+      font-weight: 100;
+      font-size: 10pt;
```
Comment (dplumb94) [inline saplings/circuits/src/components/circuitsTable/CircuitsTable.scss]: use rems for font size

File: saplings/circuits/src/components/circuitsTable/TableHeader.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { Circuit } from '../../data/processCircuits';
+
+const sortCircuits = (circuits, action) => {
```
Comment (dplumb94) [inline saplings/circuits/src/components/circuitsTable/TableHeader.js]: I would recommend moving this logic into the circuits reducer, since the logic of sorting really doesnt have anything to do with the header specifically.

File: saplings/circuits/src/components/circuitsTable/TableRow.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <tr className="table-row">
+      <td
+        className={
+          circuit.comments === 'N/A'
+            ? 'ellipsis-overflow text-grey'
+            : 'ellipsis-overflow'
+        }
+      >
+        {circuit.comments}
```
Comment (dplumb94) [inline saplings/circuits/src/components/circuitsTable/TableRow.js]: The circuit comment could probably wrap to take up more of the available space instead of cutting off.

File: saplings/circuits/src/components/circuitsTable/TableRow.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  );
+};
+
+const TableRow = ({ circuit }) => {
+  const nodeID = useLocalNodeState();
+  return (
+    <tr className="table-row">
+      <td
+        className={
+          circuit.comments === 'N/A'
```
Comment (dplumb94) [inline saplings/circuits/src/components/circuitsTable/TableRow.js]: This could be up for discussion, but I would expect to see comments on the far right column, and the ID on the first column on the left.

**Review by dplumb94 — CHANGES_REQUESTED** [review]

File: saplings/circuits/src/components/circuitsTable/CircuitsTable.scss (hunk 2 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/CircuitsTable.scss]: updated

File: saplings/circuits/src/components/circuitsTable/TableHeader.js (hunk 3 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: I don't know. I think it makes sense more in the header, because the circuits table, and the header in particular are the only components that have knowledge of what columns are available for sorting. For me it does not make sense to leak that knowledge to the circuits reducer that is defined in the parent.

File: saplings/circuits/src/components/circuitsTable/TableRow.js (hunk 4 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableRow.js]: updated

File: saplings/circuits/src/components/circuitsTable/TableRow.js (hunk 5 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableRow.js]: updated

File: app/src/theme/colors.scss (hunk 1 above)
Comment (eloaverona) [reply app/src/theme/colors.scss:33]: I will create a PR for canopy, but will leave this value for now

File: saplings/circuits/src/components/circuitsTable/TableHeader.js [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+      return sorted;
+    }
+    default:
+      return circuits;
+  }
+};
+
+const TableHeader = ({ dispatch, circuits }) => {
+  const [sorted, setSortedAsc] = useState({ asc: false, field: '' });
```
Comment (dplumb94) [inline saplings/circuits/src/components/circuitsTable/TableHeader.js]: I think the naming on these consts is a bit confusing. The typical convention is to have `[x, useX]`, so it is very clear what is being updated. I'm also not 100% sure what 'asc' is, so just spelling it out probably wouldn't hurt for maximum clarity.

File: saplings/circuits/src/components/circuitsTable/TableHeader.js (hunk 3 above)
Comment (dplumb94) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: I see your point. However, since we are using that reducer to manage the state of our circuits, I would say that it does have knowledge of what properties each circuit has. There isn't anything wrong with crafting that reducer in a way that suits our needs. 

When we are dispatching an action from a view, we are telling it to update the state in some way based on the keyword.  We also have the option to include some other data in the action's payload, but I think that requiring the view component to provide the logic for updating state would be too far. That should be the responsibility of the reducer.

If you move this functionality (and possibly the filter reducer) to the Content component, or even a different module, I think that would improve the separation of concerns in this part of the app.

File: saplings/circuits/src/components/circuitsTable/TableHeader.js (hunk 6 above)
Comment (dplumb94) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: It would probably make sense to move this to the product reducer as well, since the reducer is really what is managing the sorted circuits.

**Review by dplumb94 — CHANGES_REQUESTED** [review]
Functionality looks good. Some arguable critiques about organization.

File: saplings/circuits/src/components/circuitsTable/TableHeader.js (hunk 3 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: Okay, I see your point. I moved everything related to managing circuits state to its own file

File: saplings/circuits/src/components/circuitsTable/TableHeader.js (hunk 6 above)
Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: The convention I noticed by reading the [React docs](https://reactjs.org/docs/hooks-reference.html#usestate) was something like: 
```
const [state, setState] = useState(initialState);
```

That said, I agree that my naming was not very clear and I updated the it. Hopefully, it's better now.

Comment (eloaverona) [reply saplings/circuits/src/components/circuitsTable/TableHeader.js]: > It would probably make sense to move this to the product reducer as well, since the reducer is really what is managing the sorted circuits.

I don't think it makes sense to move this part to the circuits reducer. I think this is more about the component internal state then really relevant to the whole circuits state.  No other component cares about how the circuits are sorted. This would only bloat the circuits reducer, I think.

**Review by dplumb94 — DISMISSED** [review]

**Review by agunde406 — DISMISSED** [review]

**Review by agunde406 — APPROVED** [review]

**Review by dplumb94 — APPROVED** [review]
